connectors/utils.py

```diff
@@ -892,3 +893,9 @@ def shorten_str(string, shorten_by):
     else:
         # keep one more at the front
         return f"{string[:keep + 1]}...{string[-keep:]}"
+
+
+def ensure_python_3_10_or_higher():
```

Comment:
I feel this can stay in `setup.py`. You added it here because you want to add some tests?